### PR TITLE
Support debug/debugserver of qemu targets with West [WIP]

### PR DESCRIPTION
**Pull request by nashif**
- cmake: fix comments re qemu targets
- west: runner: add support for qemu/qemu_flags
- west: add support for debug/debugserver with qemu

Fixes #35366

### CODE REVIEW

**Comment by nashif** [comment]
works in general, but my terminal get corrupted when connecting to GDB...

File: scripts/west_commands/runners/qemu.py [hunk 1]
```diff
@@ -4,12 +4,28 @@
 
 '''Runner stub for QEMU.'''
 
+from pathlib import Path
 from runners.core import ZephyrBinaryRunner, RunnerCaps
 
 
+DEFAULT_KERNEL="zephyr.elf"
```
Comment (mbolivar-nordic) [inline scripts/west_commands/runners/qemu.py:11]: There's already an --elf-file argument; please just use that instead.

File: boards/arm/qemu_cortex_r5/board.cmake [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 set(EMU_PLATFORM qemu)
 set(QEMU_ARCH xilinx-aarch64)
+board_set_flasher_ifnset(qemu)
+board_set_debugger_ifnset(qemu)
```
Comment (mbolivar-nordic) [inline boards/arm/qemu_cortex_r5/board.cmake:8]: Just add a boards/common/qemu.board.cmake which includes these lines, and include it from the board.cmake files, the same way we do for the other runners.

**Review by mbolivar-nordic — COMMENTED** [review]
Nice! Some early feedback. Thanks for taking this on.

**Review by tejlmand — COMMENTED** [review]
If addressing the comments from @mbolivar-nordic, then I think this looks good.

**Pull request closed without merging**